tools/fpgaconf/fpgaconf.c

```diff
@@ -273,7 +273,7 @@ int parse_args(int argc, char *argv[])
 				break;
 			endptr = NULL;
 			config.target.bus = (int) strtoul(tmp_optarg, &endptr, 0);
-			if (endptr != tmp_optarg + strlen(tmp_optarg)) {
+			if (endptr != tmp_optarg + strnlen_s(tmp_optarg, sizeof(tmp_optarg))) {
```

Comment:
How about the following faster alternative that doesn't need `strlen()` at all:

    if (*tmp_optarg == '\0' || *endptr != '\0') {
        fprintf(...)
        ...

This would treat the bus argument as invalid if it's either empty (`tmp_optarg` points to '\0') or invalid (`endptr` points to something other that a '\0').

Actually, the existing code would accept an empty argument and set the bus to `0`, so

    if (*endptr != '\0') {
        fprintf(...)
        ...

would be enough. The check for an empty argument doesn't hurt, though.